benchmarks/lib/src/jmh/java/com/amazon/corretto/crypto/provider/benchmarks/AesBase.java

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider.benchmarks;
+
+import java.security.Key;
+import java.security.spec.AlgorithmParameterSpec;
+import javax.crypto.Cipher;
+import javax.crypto.spec.SecretKeySpec;
+
+public abstract class AesBase {
+  protected static final int PLAINTEXT_SIZE = 1024 * 1024;
+
+  protected Key key;
+  protected AlgorithmParameterSpec params1;
+  protected AlgorithmParameterSpec params2;
+  protected Cipher encryptor;
+  protected Cipher decryptor;
+  protected byte[] plaintext;
+  protected byte[] ciphertext;
+
+  protected abstract String getMode();
+
+  protected abstract AlgorithmParameterSpec createParameterSpec(byte[] iv);
+
+  protected abstract int getIvSize();
+
+  protected void setup(int keyBits, String provider, String padding) throws Exception {
+    BenchmarkUtils.setupProvider(provider);
+    key = new SecretKeySpec(BenchmarkUtils.getRandBytes(keyBits / 8), "AES");
+    params1 = createParameterSpec(BenchmarkUtils.getRandBytes(getIvSize()));
+    params2 = createParameterSpec(BenchmarkUtils.getRandBytes(getIvSize()));
+    final String algorithm = "AES/" + getMode() + "/" + padding;
```

Comment:
nit: `padding` and `mode` should be wired up in the same way.

Reply:
Reason for this in description:

> We also considered marking cipher mode (i.e. GCM, CBC, or CFB) a benchmark parameter instead of distinct benchmark classes, but declined to do so for similar reasons.